pkg/query/querier.go

```diff
@@ -92,72 +92,75 @@ func newQueryableCreator(
 			gateProviderFn: func() gate.Gate {
 				return gf.New()
 			},
-			maxConcurrentSelects:                maxConcurrentSelects,
-			selectTimeout:                       selectTimeout,
-			shardInfo:                           shardInfo,
-			seriesStatsReporter:                 seriesStatsReporter,
-			groupReplicaPartialResponseStrategy: groupReplicaPartialResponseStrategy,
+			maxConcurrentSelects: maxConcurrentSelects,
+			selectTimeout:        selectTimeout,
+			shardInfo:            shardInfo,
+			seriesStatsReporter:  seriesStatsReporter,
+			opts:                 opts,
 		}
 	}
 }
 
+type Options struct {
+	GroupReplicaPartialResponseStrategy bool
+	EnableDedupMerge                    bool
+}
```

Comment:
This is much better than a simple var! We can add more options in the future.

Reply:
yep